hazelcast/src/test/java/com/hazelcast/client/map/ClientMapTest.java

```diff
@@ -380,6 +390,58 @@ public void testAsyncPutAll()  {
         assertEquals(tmpMap, new HashMap<>(map));
     }
 
+    @Test
+    public void testPutAllWithMetadata() {
+        String mapName = randomMapName();
+        client.getConfig().addMapConfig(new MapConfig(mapName).setPerEntryStatsEnabled(true));
+        IMap<String, String> map = client.getMap(mapName);
+        ClientMapProxy<String, String> mapProxy = (ClientMapProxy<String, String>) map;
+
+        // The times are in milliseconds, but have second precision
+        long now = Duration.ofSeconds(Instant.now().getEpochSecond()).toMillis();
+        SimpleEntryView<String, String> entryView = new SimpleEntryView<String, String>()
+                .withKey("key")
+                .withValue("value")
+                .withCreationTime(now)
+                .withHits(42)
+                .withExpirationTime(now + Duration.ofMinutes(1).toMillis())
+                .withLastAccessTime(now)
+                .withLastUpdateTime(now)
+                .withVersion(5)
+                .withTtl(60_000)
+                .withMaxIdle(60_000);
+
+        CompletableFuture<Void> future = mapProxy.putAllWithMetadataAsync(newArrayList(entryView));
```

Comment:
Can you please use the new API with multiple entries and test?

Reply:
10965d8fc60